### PR TITLE
Remove obsolete deletions and document post_apply

### PR DESCRIPTION
**Pull request by linki**
As suggested in https://github.com/zalando-incubator/kubernetes-on-aws/pull/574 this removes unneccessary deletions.

### CODE REVIEW

**Comment by linki** [comment]
👍

**Comment by hjacobs** [comment]
:+1:

File: cluster/manifests/deletions.yaml [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
-  namespace: kube-system
-  kind: daemonset
+# example: remove the app-ingress-controller deployment from kube-system
+# pre_apply:
+# - name: app-ingress-controller
+#   namespace: kube-system
+#   kind: deployment
+
+pre_apply: {} # everything defined under here will be deleted before applying the manifests
+post_apply: {} # everything defined under here will be deleted after applying the manifests
```
Comment (mikkeloscar) [inline cluster/manifests/deletions.yaml]: The value is a list not a map :)

Comment (linki) [reply cluster/manifests/deletions.yaml]: Fixed and tested this time :)

**Comment by linki** [comment]
👍

**Comment by linki** [comment]
Good to go!

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**